### PR TITLE
upgrade log4j2 to 2.17.1

### PR DESCRIPTION
**Pull request by jrhizor**
There were problems upgrading: https://github.com/airbytehq/airbyte/pull/8969

I'm struggling to figure out why this isn't working properly. Even without any changes to the xml config, the original PR was working for acceptance tests on Docker (but failing on Kubernetes).

Then, if you remove the system property lookups, it still works on Docker and _partially_ works on Kubernetes. It shows the logs but the container highlighting isn't visible in the UI (while it is in worker logs).

As far as I can tell the appender only updates the underlying log4j dependency in the upgrade (see https://github.com/bluedenim/log4j-s3-search/pull/118/files).

Reading https://www.zdnet.com/article/apache-releases-new-2-17-0-patch-for-log4j-to-solve-denial-of-service-vulnerability/ or https://logging.apache.org/log4j/2.x/changes-report.html#a2.17.0 doesn't show anything particularly strange (maybe I'm missing something).

@davinchia any idea why this would have different behavior on Kubernetes?

---
This change is [Reviewable](https://reviewable.io/reviews/airbytehq/airbyte/8977)

### CODE REVIEW

File: build.gradle
Comment (0001vrn) [inline build.gradle:279]: It'd be better if we use the `log4j2.version` property and replace `2.17.0` references.

**Comment by davinchia** [comment]
Hmm looks like a config error - will look after I get the bootloader changes done.

**Review by lmossman — APPROVED** [review]
It seems worth it to me to do this upgrade to resolve the security vulnerability even if it makes our logs highlighting not work. Maybe we should create a separate issue for that?

**Comment by jrhizor** [comment]
No user-submitted values make it into our context lookups (it's env variables or things like file paths we generate off of ids, not user submitted values), so I don't think we're at risk from abuse from this bug. I think we have time to fix this separately.

**Comment by davinchia** [comment]
@jrhizor I can find time to look at this tomorrow.

The one thing that came to mind about different behaviour - we are using `therealvan` plugin on Kubernetes. Perhaps related to that.

**Comment by davinchia** [comment]
I see these colors in the logs:

Screen Shot 2021-12-22 at 7 25 16 PM

**Comment by davinchia** [comment]
The Kube acceptance test is passing for me locally.

**Comment by davinchia** [comment]
@jrhizor I think we were passing the wrong ANSI codes earlier. I switched them to colors instead of background and got it working.

Screen Shot 2021-12-22 at 10 34 59 PM

File: airbyte-commons/src/main/resources/log4j2.xml
Comment (davinchia) [inline airbyte-commons/src/main/resources/log4j2.xml:22]: @jrhizor looks like we were using the wrong syntax here.

**Comment by davinchia** [comment]
@jrhizor this should be good. I'll let you merge it in!

**Review by davinchia — APPROVED** [review]

**Comment by jrhizor** [comment]
@davinchia it was always working in the cli logs, but it was broken in the UI only. I can double check, maybe backgrounds were specifically failing in the UI, but only after this change? Backgrounds were working properly before.

File: airbyte-commons/src/main/resources/log4j2.xml
Comment (jrhizor) [reply airbyte-commons/src/main/resources/log4j2.xml:22]: Double means it's evaluated at log time instead of config time, which is necessary for the sys vars I believe (maybe this is broken in the upgrade somehow)

**Comment by jrhizor** [comment]
As-is there's no highlighting on Kubernetes still. Was it working for you @davinchia after making your changes here?

**Comment by cavanaug** [comment]
When is this going to get merged???   This is a security issue and at least within our organization, our systems are getting shutdown if log4j < 2.17 is detected?

**Comment by lmossman** [comment]
@jrhizor I just tried merging master into this branch and ran it locally, and it looks like the log highlighting is working now. I pushed the merge and I also switch the color settings back to what they were originally since that wasnt the issue, and the new text colors were hard to read on a white background.

Here is a screenshot of me running this locally with the new merge from master:
Screen Shot 2022-02-03 at 5 29 47 PM

So I think this should be safe to merge!

**Review by 0001vrn — APPROVED** [review]

**Comment by jrhizor** [comment]
I'm removing the fallback logic from system property to environment variable entirely. For the container orchestrator I'll just specify a completely new xml config for log4j.

I'm going to re-run this locally after that change to verify that formatting is working across docker + kube. If it is I'll merge right away.

**Comment by jrhizor** [comment]
It's working in both envs. Merging.

**Comment by cavanaug** [comment]
Thank you!!!

**Pull request merged**
